### PR TITLE
Divyanshu | Test | Implementation of Zimbabwe VAT in GIDDH 

### PR DESCRIPTION
**Pull request by dvCodeWorld**
Zimbabwe - https://app.clickup.com/t/86cuy39kd

### CODE REVIEW

File: apps/web-giddh/src/app/services/vat.service.ts [hunk 1]
```diff
@@ -69,10 +74,11 @@ export class VatService {
             }), catchError((e) => this.errorHandler.HandleCatch<any, any>(e, request)));
     }
 
-    public downloadVatReport(request: VatReportRequest): Observable<BaseResponse<any, any>> {
+    public downloadVatReport(request: VatReportRequest, countryCode: 'UK' | 'ZW'): Observable<BaseResponse<any, any>> {
         this.companyUniqueName = this.generalService.companyUniqueName;
+        const api = countryCode === 'UK' ? VAT_API.DOWNLOAD_REPORT : VAT_API.DOWNLOAD_ZW_REPORT;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/services/vat.service.ts]:
```suggestion
        const apiEndPoint = countryCode === 'UK' ? VAT_API.DOWNLOAD_REPORT : VAT_API.DOWNLOAD_ZW_REPORT;
```

File: apps/web-giddh/src/app/services/vat.service.ts [hunk 2]
```diff
@@ -24,14 +24,19 @@ export class VatService {
      * @return {*}  {Observable<BaseResponse<any, any>>}
      * @memberof VatService
      */
-    public getUKVatReport(request: VatReportRequest): Observable<BaseResponse<any, any>> {
+    public getCountryWiseVatReport(request: VatReportRequest, countryCode: 'UK' | 'ZW' = 'UK'): Observable<BaseResponse<any, any>> {
         this.companyUniqueName = this.generalService.companyUniqueName;
 
-        let url = this.config.apiUrl + VAT_API.VIEW_REPORT;
+        const api = countryCode === 'UK' ? VAT_API.VIEW_REPORT : VAT_API.VIEW_ZW_REPORT;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/services/vat.service.ts]:
```suggestion
        const apiEndPoint = countryCode === 'UK' ? VAT_API.VIEW_REPORT : VAT_API.VIEW_ZW_REPORT;
```

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/services/vat.service.ts (hunk 1 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/services/vat.service.ts]: apiEndPoint

File: apps/web-giddh/src/app/services/vat.service.ts (hunk 2 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/services/vat.service.ts]: apiEndPoint

**Review by dilpreetsinghofficial — APPROVED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**